packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -66,6 +80,123 @@ class TextMapPropagator {
     })
   }
 
+  _injectB3 (spanContext, carrier) {
+    if (!this._config.experimental.b3) return
+
+    carrier[b3TraceKey] = spanContext._traceId.toString('hex')
+    carrier[b3SpanKey] = spanContext._spanId.toString('hex')
+    carrier[b3SampledKey] = spanContext._traceFlags.sampled ? '1' : '0'
+
+    if (spanContext._parentId) {
+      carrier[b3ParentKey] = spanContext._parentId.toString('hex')
+    }
+  }
+
+  _extractSpanContext (carrier) {
+    const context = this._extractContext(carrier)
+
+    if (!context) return null
+
+    if (context.traceFlags.sampled !== false) {
+      return new DatadogSpanContext(context)
+    } else {
+      return new NoopSpanContext(context)
+    }
+  }
+
+  _extractContext (carrier) {
+    return this._extractDatadogContext(carrier) || this._extractB3Context(carrier)
+  }
+
+  _extractDatadogContext (carrier) {
+    const sampled = this._isSampled(carrier[sampleKey])
+
+    return this._extractGenericContext(carrier, traceKey, spanKey, sampled, 10)
+  }
+
+  _extractB3Context (carrier) {
+    if (!this._config.experimental.b3) return null
+
+    const b3 = this._extractB3Headers(carrier)
+    const sampled = this._isSampled(b3[b3SampledKey], b3[b3FlagsKey] === '1')
```

Comment:
Let's just pass in `b3[b3FlagsKey]` and let `_isSampled` handle the check. This would be consistent with how we're passing in `b3[b3SampledKey]`.

Reply:
We now use `debug` outside the function, so let's leave it to not duplicate the logic.